eventstore/tasks.py

```diff
@@ -119,3 +130,125 @@ def delete_rapidpro_contact_by_uuid(contact_uuid):
     | delete_rapidpro_contact_by_uuid.s()
 )
+
+
+@app.task(
+    autoretry_for=(RequestException, SoftTimeLimitExceeded),
+    retry_backoff=True,
+    max_retries=1,
+    acks_late=True,
+    soft_time_limit=10,
+    time_limit=15,
+)
+def get_rapidpro_contact_by_urn(urn):
+    if not urn:
+        return
+    return rapidpro.get_contacts(urn=urn).first(retry_on_rate_exceed=True).serialize()
+
+
+@app.task(
+    autoretry_for=(RequestException, SoftTimeLimitExceeded),
+    retry_backoff=True,
+    max_retries=1,
+    acks_late=True,
+    soft_time_limit=10,
+    time_limit=15,
+)
+def check_contact_timestamp(contact):
+    timestamp = contact["fields"].get("whatsapp_undelivered_timestamp")
+    current_date = get_utc_now()
+
+    send = False
+    if timestamp:
+        last_date = dateparse.parse_datetime(timestamp)
+        if (current_date - last_date).days >= settings.WHATSAPP_EXPIRY_SMS_BOUNCE_DAYS:
+            send = True
+    else:
+        send = True
+
+    language = contact.get("language", "eng").lower()
+    context = {"language": f"{language}-ZA"}
+    if send:
+        try:
+            _, msisdn = contact["urns"][0].split(":")
+            context["msisdn"] = msisdn.lstrip("+")
+        except (KeyError, IndexError, ValueError, AttributeError):
+            pass
+    else:
+        pass
+
+    return context
+
+
+@app.task(
+    autoretry_for=(RequestException, SoftTimeLimitExceeded),
+    retry_backoff=True,
+    max_retries=15,
+    acks_late=True,
+    soft_time_limit=10,
+    time_limit=15,
+)
+def send_undelivered_sms(context):
+    if "msisdn" not in context:
+        return {}
+
+    headers = {
+        "Authorization": "Bearer {}".format(settings.TURN_TOKEN),
+        "Content-Type": "application/json",
+        "x-turn-fallback-channel": "1",
+    }
+
+    with translation.override(context["language"]):
+        text = translation.ugettext(
+            "We see that your MomConnect WhatsApp messages are not being "
+            "delivered. If you would like to receive your messages over "
+            "SMS, reply ‘SMS’."
+        )
+
+    data = json.dumps(
+        {
+            "preview_url": False,
+            "recipient_type": "individual",
+            "to": context["msisdn"],
+            "type": "text",
+            "text": {"body": text},
+        }
+    )
+
+    r = requests.post(
+        urljoin(settings.TURN_URL, "v1/messages"), headers=headers, data=data
+    )
+    r.raise_for_status()
+
+    return context
+
+
+@app.task(
+    autoretry_for=(RequestException, SoftTimeLimitExceeded),
+    retry_backoff=True,
+    max_retries=15,
+    acks_late=True,
+    soft_time_limit=10,
+    time_limit=15,
+)
+def update_rapidpro_contact_error_timestamp(context):
+    if "msisdn" not in context:
+        return
+
+    msisdn = context["msisdn"]
+    rapidpro.update_contact(
+        f"whatsapp:{msisdn}",
+        fields={
+            "whatsapp_undelivered_timestamp": get_utc_now().strftime(
```

Comment:
We can probably just use `get_utc_now().isoformat()` to get the ISO formatted datetime string.